WordPress/Sniffs/VIP/SlowDBQuerySniff.php

```diff
@@ -50,6 +51,10 @@ public function getGroups() {
 	 */
 	public function process_token( $stackPtr ) {
 
+		if ( $this->has_whitelist_comment( 'slow query', $stackPtr ) ) {
+			return;
+		}
+
 		if ( $this->has_whitelist_comment( 'tax_query', $stackPtr ) ) {
 			return;
```

Comment:
@barryceelen I'll have a look when I'm back home if you like.

```php
$this->addMessage( $message, $stackPtr, false );
```

You can just use a straight `$this->phpcsFile->addWarning()` for this.
Also: please add a distinct error code to the message, something like `DeprecatedWhitelistFlagFound`